public/components/model_version/__tests__/version_artifact.test.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { render, screen } from '../../../../test/test_utils';
+import { ModelVersionFormData } from '../types';
+import { ModelVersionArtifact } from '../version_artifact';
+
+const DEFAULT_VALUES = {
+  artifactSource: 'source_not_changed' as const,
+  modelURL: 'http://url.to/artifact.zip',
+  modelFileFormat: 'TORCH_SCRIPT',
+  configuration: '{}',
+};
+
+const TestApp = ({ defaultValues = DEFAULT_VALUES }: { defaultValues?: ModelVersionFormData }) => {
+  const form = useForm({
+    defaultValues,
+  });
+
+  return (
+    <FormProvider {...form}>
+      <ModelVersionArtifact />
+    </FormProvider>
+  );
+};
+
+describe('<ModelVersionArtifact />', () => {
+  it('should display a readonly "Artifact and configuration" panel', () => {
+    render(<TestApp />);
+    expect(screen.getByLabelText('Uploaded artifact details(URL)')).toHaveAttribute('readonly');
+    expect(screen.getByLabelText('Model file format')).toHaveAttribute('readonly');
+    expect(screen.getByLabelText('readonly configuration')).toBeInTheDocument();
+
+    expect(screen.getByDisplayValue('http://url.to/artifact.zip')).toBeInTheDocument();
+    expect(screen.getByDisplayValue('Torchscript(.pt)')).toBeInTheDocument();
+  });
+
+  it('should display a readonly input with file name', () => {
+    const defaultValues = {
+      artifactSource: 'source_not_changed' as const,
+      modelFile: new File([], 'artifact.zip'),
+      modelFileFormat: 'TORCH_SCRIPT',
+      configuration: '{}',
+    };
+    render(<TestApp defaultValues={defaultValues} />);
+    expect(screen.getByDisplayValue('artifact.zip')).toBeInTheDocument();
+    expect(screen.getByDisplayValue('artifact.zip')).toHaveAttribute('readonly');
+  });
+
+  it('should display "Artifact source select" after clicking "Edit" button', async () => {
+    const user = userEvent.setup();
+    render(<TestApp />);
+
+    await user.click(screen.getByLabelText('edit version artifact'));
+    expect(screen.getByText('Artifact source')).toBeInTheDocument();
+    expect(screen.getByLabelText('Upload new from local file')).toBeInTheDocument();
+    expect(screen.getByLabelText('Upload new from URL')).toBeInTheDocument();
+  });
+
+  it('should select "Keep existing" by default after click "Edit" button', async () => {
+    const user = userEvent.setup();
+    render(<TestApp />);
+
+    await user.click(screen.getByLabelText('edit version artifact'));
+    expect(screen.getByLabelText('Keep existing')).toBeChecked();
+    // model url and model file format are readonly
+    expect(screen.getByDisplayValue('http://url.to/artifact.zip')).toHaveAttribute('readonly');
+    expect(screen.getByDisplayValue('Torchscript(.pt)')).toHaveAttribute('readonly');
```

Comment:
From my perspective, the model file format should be editable. Then user can modify the model file format without re-upload model. Let's check with designer about that. Won't block PR approve.

Reply:
You're right, PR updated